mqtt_exporter/main.py

```diff
@@ -326,14 +326,21 @@ def _parse_message(raw_topic, raw_payload):
     """Parse topic and payload to have exposable information."""
     # parse MQTT payload
     try:
-        payload = json.loads(raw_payload)
-    except json.JSONDecodeError:
-        LOG.debug('failed to parse payload as JSON: "%s"', raw_payload)
-        return None, None
+        if not isinstance(raw_payload, str):
+            raw_payload = raw_payload.decode(json.detect_encoding(raw_payload))
     except UnicodeDecodeError:
         LOG.debug('encountered undecodable payload: "%s"', raw_payload)
         return None, None
 
+    if raw_payload in STATE_VALUES:
+        payload = STATE_VALUES[raw_payload]
+    else:
+        try:
+            payload = json.loads(raw_payload)
```

Comment:
if the object is a non utf-8 string, it will crash as the UnicodeDecodeError is not catched anymore here.

```
>>> a = "\x81"
>>> isinstance(a, str)
True
>>> a.decode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'. Did you mean: 'encode'?
```

Reply:
If it was a non utf-8 string, it will be decoded on line 330 into a str. That's where the UnicodeDecodeError can happen, and it is handled there. After that line, a UnicodeDecodeError can't happen.